### PR TITLE
Refactor build loop a bit

### PR DESCRIPTION
**Pull request by Profpatsch**
This is the first step to solving https://github.com/target/lorri/issues/341, which is refactoring the build loop. See the commits, there should be no changes in functionality in this code.

cc @curiousleo
with help from @NinjaTrappeur

### CODE REVIEW

File: src/build_loop.rs [hunk 1]
```diff
@@ -95,66 +94,91 @@ impl<'a> BuildLoop<'a> {
             }
         };
 
+        let mut output_paths;
         // The project has just been added, so run the builder in the first iteration
-        let mut reason = Some(Event::Started {
+        let reason = Event::Started {
```
Comment (curiousleo) [inline src/build_loop.rs]: I would inline `reason` (used once) to remove one variable from the scope.

File: src/build_loop.rs [hunk 2]
```diff
@@ -95,66 +94,91 @@ impl<'a> BuildLoop<'a> {
...
             nix_file: self.project.nix_file.clone(),
             reason: Reason::ProjectAdded(self.project.nix_file.clone()),
-        });
-        let mut output_paths = None;
+        };
+        output_paths = self.once_with_send(&tx, reason);
```
Comment (curiousleo) [inline src/build_loop.rs]:
```suggestion
        let mut output_paths = self.once_with_send(&tx, reason);
```

File: src/build_loop.rs [hunk 3]
```diff
@@ -95,66 +94,91 @@ impl<'a> BuildLoop<'a> {
             }
         };
 
+        let mut output_paths;
```
Comment (curiousleo) [inline src/build_loop.rs]: No need for this separate declaration.

**Comment by curiousleo** [comment]
>  Fix build loop channel debounce
Would you mind changing the title of this PR? It's out of sync with the description, which makes it clear that this is a preparatory refactor that does not actually change behaviour.

**Comment by Profpatsch** [comment]
> Would you mind changing the title of this PR? It's out of sync with the description, which makes it clear that this is a preparatory refactor that does not actually change behaviour.

yeah, that’s just github expanding the branch name :) Changed

**Pull request merged**